MapboxNavigation/SystemSpeechSynthesizer.swift

```diff
@@ -0,0 +1,172 @@
+
+import AVFoundation
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxSpeech
+
+/**
+ `SpeechSynthesizing` implementation, using `AVSpeechSynthesizer`. Supports only english language.
+ */
+open class SystemSpeechSynthesizer: NSObject, SpeechSynthesizing {
+    
+    // MARK: - Properties
+    
+    public weak var delegate: SpeechSynthesizingDelegate?
+    public var muted: Bool = false {
+        didSet {
+            if isSpeaking {
+                interruptSpeaking()
+            }
+        }
+    }
+    public var volume: Float {
+        get {
+            return NavigationSettings.shared.voiceVolume
+        }
+        set {
+            // Do Nothing
+            // AVSpeechSynthesizer uses 'AVAudioSession.sharedInstance().outputVolume' by default
+        }
+    }
+    public var isSpeaking: Bool { return speechSynth.isSpeaking }
+    public var locale: Locale = Locale.autoupdatingCurrent
+    
+    private var speechSynth: AVSpeechSynthesizer
+    private var previousInstrcution: SpokenInstruction?
+    
+    // MARK: - Lifecycle
+    
+    override public init() {
+        speechSynth = AVSpeechSynthesizer()
+        super.init()
+        speechSynth.delegate = self
+    }
+    
+    deinit {
+        interruptSpeaking()
+    }
+    
+    // MARK: - Public methods
+    
+    open func prepareIncomingSpokenInstructions(_ instructions: [SpokenInstruction]) {
+        // Do nothing
+    }
+    
+    open func speak(_ instruction: SpokenInstruction, during legProgress: RouteLegProgress) {
+        guard !muted else {
+            delegate?.speechSynthesizer(self,
+                                        didSpeak: instruction,
+                                        with: nil)
+            return
+        }
+        
+        var utterance: AVSpeechUtterance?
+        if Locale.preferredLocalLanguageCountryCode == "en-US" {
+            // Alex can’t handle attributed text.
+            utterance = AVSpeechUtterance(string: instruction.text)
+            utterance!.voice = AVSpeechSynthesisVoice(identifier: AVSpeechSynthesisVoiceIdentifierAlex)
+        }
+        
+        let modifiedInstruction = delegate?.speechSynthesizer(self, willSpeak: instruction) ?? instruction
+        
+        if utterance?.voice == nil {
+            utterance = AVSpeechUtterance(attributedString: modifiedInstruction.attributedText(for: legProgress))
+        }
+        
+        // Only localized languages will have a proper fallback voice
+        if utterance?.voice == nil {
+            utterance?.voice = AVSpeechSynthesisVoice(language: Locale.preferredLocalLanguageCountryCode)
```

Comment:
I think we should also try to pass in the `speechLocale` here too. This code has used `Locale.preferredLocalLanguageCountryCode` since #187, even before we moved guidance instruction generation to the server side, so I think this is buggy behavior that we’ve overlooked for a long time.